cdc/generation.cc

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) 2019 ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <boost/type.hpp>
+
+#include "keys.hh"
+#include "bytes_ostream.hh"
+#include "serializer.hh"
+#include "serializer_impl.hh"
+#include "dht/i_partitioner.hh"
+
+#include "cdc/generation.hh"
+
+#include "idl/token.dist.hh"
+#include "idl/token.dist.impl.hh"
+#include "idl/cdc_topology_description.dist.hh"
+#include "idl/cdc_topology_description.dist.impl.hh"
+
+namespace cdc {
+
+stream_id::stream_id()
+    : _first(-1), _second(-1) {}
```

Comment:
Is this useful? If we always make new stream ids, better not leave an option for creating an invalid one.

Reply:
I need the default constructor for the generation of streams algorithm (`generate_topology_description`). I guess I could use `std::optional` there, but I'm afraid it would hurt performance, and it would probably make the code less readable.

Having such a default constructor doesn't hurt. We're generating these ids randomly. The probability of hitting this particular pair of integers is practically zero